### ISSUE TITLE
pkgconfig test broken on CentOS 7 / Boost 1.61

### ISSUE THREAD

**Issue by zao** [STEllAR-GROUP/hpx#2242]
The tests for pkgconfig files (installed and build) fail on my CentOS 7 machine, missing symbols from `libdl.so`:
```
/usr/bin/ld: /tmp/cc9gF3u7.o: undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
/usr/bin/ld: note: 'dlclose@@GLIBC_2.2.5' is defined in DSO /lib64/libdl.so.2 so try adding it to the linker command line
/lib64/libdl.so.2: could not read symbols: Invalid operation
collect2: error: ld returned 1 exit status
```
The generated `hpx_application.pc` looks like this:
`Libs: -std=c++11 -L${libdir} -L ${libdir}/libhpx.so ${libdir}/libhpx_init.a  -ldl -lrt /home/zao/boost-1.61-gcc48/lib/libboost_chrono.so /home/zao/boost-1.61-gcc48/lib/libboost_date_time.so /home/zao/boost-1.61-gcc48/lib/libboost_filesystem.so /home/zao/boost-1.61-gcc48/lib/libboost_program_options.so /home/zao/boost-1.61-gcc48/lib/libboost_regex.so /home/zao/boost-1.61-gcc48/lib/libboost_system.so /home/zao/boost-1.61-gcc48/lib/libboost_thread.so /home/zao/boost-1.61-gcc48/lib/libboost_atomic.so /usr/lib64/libpthread.so /home/zao/boost-1.61-gcc48/lib/libboost_context.so /home/zao/boost-1.61-gcc48/lib/libboost_thread.so /home/zao/boost-1.61-gcc48/lib/libboost_chrono.so /home/zao/boost-1.61-gcc48/lib/libboost_system.so /home/zao/boost-1.61-gcc48/lib/libboost_date_time.so /home/zao/boost-1.61-gcc48/lib/libboost_atomic.so /home/zao/boost-1.61-gcc48/lib/libboost_random.so /home/zao/boost-1.61-gcc48/lib/libboost_system.so /home/zao/boost-1.61-gcc48/lib/libboost_atomic.so /usr/lib64/libtcmalloc_minimal.so /usr/lib64/libhwloc.so`

Hacking in `-ldl` last on the lines makes the link succeed, so I assume the dependency is from the Boost libraries mentioned after the first occurence of `-ldl`.

**Comment by hkaiser** [comment]
That should have been fixed by now (see 748991db65b8ec324ca39d137d7135cacd27d580). Please pull from master and try again. It was caused by the 'empty' `-L` option in the pkgconfig file.

**Comment by zao** [comment]
I can verify that `master` solves the problem.
